### ISSUE TITLE
Improve/update information about TYPO3 PHP coding standards

### ISSUE THREAD

**Issue by schams-net** [TYPO3-Documentation/TYPO3CMS-Reference-CoreApi#4011]
The TYPO3 PHP coding standards are *based* on the [PSR-12 standard](https://www.php-fig.org/psr/psr-12/) and (as far as I understand) aim to move towards the [PER-CS2.0 standard](https://www.php-fig.org/per/coding-style/).

The TYPO3 documentation mentions [PSR-2 standard](https://www.php-fig.org/psr/psr-2/) which is deprecated as of August 2019.

My understanding is that using the [TYPO3 Coding Standards package](https://github.com/TYPO3/coding-standards) is best practice, plus the config for php-cs-fixer, see <https://github.com/TYPO3/typo3/blob/main/Build/php-cs-fixer/config.php>.

**Comment by garvinhicking** [comment]
I've now created these two PRs:

https://github.com/TYPO3-Documentation/TYPO3CMS-Guide-ContributionWorkflow/pull/329
https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/4016

(the latter being backportable)

It's a bit hard due to all the cross-referencing and because the Coding Standards actually says "you gotta catch em all" and covers PHP-FIG, PSR-1/PSR-2 and PSR-12 plus maybe PER-CS2.0 not so far-off. So I tried to phrase it so that the "source of truth" is referended in t3coreapi:cgl at best.

@linawolf @brotkrueml Waiting for your feedback on this before I merge anything.
